purchases/src/main/kotlin/com/revenuecat/purchases/google/usecase/BillingClientUseCase.kt

```diff
@@ -96,4 +110,45 @@ internal abstract class BillingClientUseCase<T>(
             ).also { errorLog(it) },
         )
     }
+
+    private fun backoffOrRetryNetworkError(
+        onError: (BillingResult) -> Unit,
+        billingResult: BillingResult,
+    ) {
+        if (backoffForNetworkErrors && retryBackoffMilliseconds < RETRY_TIMER_MAX_TIME_MILLISECONDS) {
+            retryWithBackoff()
+        } else if (!backoffForNetworkErrors && retryAttempt < maxRetries) {
+            retryAttempt++
+            executeAsync()
+        } else {
+            onError(billingResult)
+        }
+    }
+
+    private fun backoffOrErrorIfUseInSession(
+        onError: (BillingResult) -> Unit,
+        billingResult: BillingResult,
+    ) {
+        if (!useCaseParams.appInBackground) {
```

Comment:
I guess this will pick the value when the request is initiated. As in, if the user backgrounds/foregrounds the app, this won't be updated. I guess this is probably ok though... I do wonder if there is a way to reset the request when this state changes, but maybe we can do that later.

Reply:
Nitpick, inverse the condition here?